doozerlib/distgit.py

```diff
@@ -766,6 +776,43 @@ def _build_container(
         self.logger.info("Successfully built image: {} ; {}".format(target_image, task_url))
         return True
 
+    def _logs_dir(self):
+        return "%s/%s" % (self.runtime.brew_logs_dir, self.metadata.name)
+
+    def _detect_permanent_build_failures(self):
+        """check logs to determine if this container build failed for a known non-flake reason"""
+        logs_dir = self._logs_dir()
+        try:
+            # find the most recent subdir, which should contain logs for the latest build task
+            task_dirs = [os.path.join(logs_dir, d) for d in os.listdir(logs_dir)]
+            task_dirs = [d for d in task_dirs if os.path.isdir(d)]
+            if not task_dirs:
+                self.logger.info("No task logs found under {}; cannot analyze logs".format(logs_dir))
+                return None
+            latest_dir = max(task_dirs, key=os.path.getmtime)
+
+            # check the log files for recognizable problems
+            found_problems = []
+            problem_regexen = [
+                # note: we read log lines, so these can only match in a single line
+                r"No package .* available",
```

Comment:
Ideally these should be in a separate file. This list will likely get very big real quick. So either a yaml file or just a python file with nothing but a list in it (I'd lean towards the latter.)

Reply:
I agree we can add that feature later if we need it, but I still want a separate file. Otherwise I fear it might turn into the same thing as "let's just put the sources in group.yaml, there aren't that many!"

Reply:
I'm thinking about other teams. Anything hard coded might not be useful to another team. Seems like these should always come from a separate input file.

Reply:
Hm. Maybe it wouldn't be hard to have this all in group.yml.